Jenkinsfile

```diff
@@ -1,57 +1,72 @@
-if (env.BRANCH_NAME == 'master') {
-    properties([
-        pipelineTriggers([
-            upstream(
-                upstreamProjects: 'ocflib/master',
-                threshold: hudson.model.Result.SUCCESS,
-            ),
-        ]),
-    ])
-}
-
+pipeline {
+  // TODO: Make this cleaner: https://issues.jenkins-ci.org/browse/JENKINS-42643
+  triggers {
+    upstream(
+      upstreamProjects: (env.BRANCH_NAME == 'master' ? 'ocflib/master' : ''),
+      threshold: hudson.model.Result.SUCCESS,
+    )
+  }
 
-try {
-    node('slave') {
-        step([$class: 'WsCleanup'])
+  agent {
+    label 'slave'
+  }
 
-        stage('check-out-code') {
-            checkout scm
-        }
-
-        stage('test') {
-            sh 'make test'
-        }
+  options {
+    ansiColor('xterm')
+    timeout(time: 1, unit: 'HOURS')
+    timestamps()
+  }
 
-        stash 'src'
+  stages {
+    stage('check-gh-trust') {
+      steps {
+        checkGitHubAccess()
+      }
     }
 
+    stage('test') {
+      steps {
+        sh 'make test'
+      }
+    }
 
-    if (env.BRANCH_NAME == 'master') {
-        node('deploy') {
-            step([$class: 'WsCleanup'])
-            unstash 'src'
-
-            stage('deploy-apps') {
-                sh 'make deploy'
-            }
+    stage('diff-apps') {
+      when {
+        not {
+          branch 'master'
         }
+      }
+      agent {
+        label 'deploy'
```

Comment:
I may be misunderstanding exactly what this does, but we want this running on a deploy agent when it's not the master branch?